controllers/storagecluster/storagecluster_controller.go

```diff
@@ -155,6 +155,53 @@ func (r *StorageClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		},
 	)
 
+	enqueueFromStorageProfile := handler.EnqueueRequestsFromMapFunc(
+		func(ctx context.Context, obj client.Object) []reconcile.Request {
+			spobj, ok := obj.(*ocsv1.StorageProfile)
+			if !ok {
+				return []reconcile.Request{}
+			}
+
+			// todo change according to standards
+			if spobj.Status.Phase != util.PhaseReady {
+				return []reconcile.Request{}
+			}
+
+			// Get the StorageCluster objects
+			scList := &ocsv1.StorageClusterList{}
+			err := r.List(r.ctx, scList, &client.ListOptions{Namespace: obj.GetNamespace()})
+			if err != nil {
+				r.Log.Error(err, "Unable to list StorageCluster objects")
+				return []reconcile.Request{}
+			}
+
+			// Return name and namespace per StorageProfile object
+			request := []reconcile.Request{}
```

Comment:
at the start of this func, if the arbitrary obj is of storageprofile type why not reconcile that single obj rather than every profile or it is just an insurance? correct me if I'm wrong here.

Reply:
Good catch, thanks